src/modules/ekf2/ekf2_main.cpp

```diff
@@ -555,6 +581,39 @@ void Ekf2::task_main()
 			if ((sensors.timestamp + sensors.magnetometer_timestamp_relative) != _timestamp_mag_us) {
 				_timestamp_mag_us = sensors.timestamp + sensors.magnetometer_timestamp_relative;
 
+				// Reset learned bias parameters if there has been a persistant change in magnetometer ID
+				// Do not reset parmameters when armed to prevent potential time slips casued by parameter set
+				// and notification events
+				// Check if there has been a persistant change in magnetometer ID
+				orb_copy(ORB_ID(sensor_selection), sensor_selection_sub, &sensor_selection);
+
+				if (sensor_selection.mag_device_id != 0 && sensor_selection.mag_device_id != _mag_bias_id.get()) {
+					if (_invalid_mag_id_count < 200) {
+						_invalid_mag_id_count++;
+					}
+
+				} else {
+					if (_invalid_mag_id_count > 0) {
+						_invalid_mag_id_count--;
+					}
+				}
+
+				if ((vehicle_status.arming_state != vehicle_status_s::ARMING_STATE_ARMED) && (_invalid_mag_id_count > 100)) {
+					// the sensor ID used for the last saved mag bias is not confirmed to be the same as the current sensor ID
+					// this means we need to reset the learned bias values to zero
+					_mag_bias_x.set(0.f);
+					_mag_bias_x.commit();
```

Comment:
Is there any reason not to use `commit_no_notification()`? (they're saved in both cases, and no other module is interested in these param values)